src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/WorkspaceDAO.scala

```diff
@@ -37,4 +43,25 @@ class FileSystemWorkspaceDAO(storageDirectory: Path) extends WorkspaceDAO {
     val json = new String(Files.readAllBytes(location)).parseJson
     WorkspaceFormat.read(json)
   }
+
+  def list(): Seq[WorkspaceShort] = {
+    val visitor = new WorkspaceFileVisitor(storageDirectory)
+    Files.walkFileTree(storageDirectory, visitor)
+    visitor.workspaces.result
+  }
 }
+
+private class WorkspaceFileVisitor(root: Path) extends SimpleFileVisitor[Path] with LazyLogging {
```

Comment:
yeah, but the java.nio api kind of forces my hand. This is how it is done if we want to use nio.